### PR TITLE
New module based on terraform-module-template

### PR DESCRIPTION
**Pull request by arman-promo**
Jira [task](https://jira.epam.com/jira/browse/EPMLDI-1601)

### CODE REVIEW

**Review by mckate — COMMENTED** [review]
h

File: README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    vpc_cidr        = "10.10.1.0/24"
+    security_groups = []
+    s3_bucket_name  = "tf-static-artifactory-storage"
+    whitelist       = []
+}
+```
+
+### Examples
+
+* [Example folder](https://github.com/lean-delivery/tf-module-aws-core/tree/master/examples)
```
Comment (mckate) [inline README.md]: wrong URL

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    "Principal": {"AWS":"${aws_cloudfront_origin_access_identity.origin_access_identity.iam_arn}"}
+    },
+    {
+    "Sid": "bucket_policy_site_all",
+    "Action": ["s3:GetObject"],
+    "Effect": "Allow",
+    "Resource": "arn:aws:s3:::${var.s3_bucket_name}/*",
+    "Principal": {"AWS":"${aws_cloudfront_origin_access_identity.origin_access_identity.iam_arn}"}
+    }
+]
```
Comment (mckate) [inline main.tf]: better move this policy into a template

File: outputs.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+output "vpc_id" {
+  value = "${module.vpc.vpc_id}"
+}
+
+output "whitelist" {
+  value = "${var.whitelist}"
```
Comment (mckate) [inline outputs.tf]: Vars must have descriptions

Comment (arman-promo) [reply outputs.tf]: [added](https://github.com/lean-delivery/tf-module-static-artifact-storage/pull/1/commits/50e11959b03f24cfae366549f194d5a75375a490)

File: main.tf (hunk 2 above)
Comment (arman-promo) [reply main.tf]: ok, [done](https://github.com/lean-delivery/tf-module-static-artifact-storage/pull/1/commits/373c84d41c978a933bb230a9249ab4e5d6582dea)

File: README.md (hunk 1 above)
Comment (arman-promo) [reply README.md]: [fixed](https://github.com/lean-delivery/tf-module-static-artifact-storage/pull/1/commits/132823d80a56a81f2f9bb0f7b8a2a3379126ef01)

File: README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## Usage
+
+### Conditional creation
+
+```
+module "static-artifact-storage" {
+    source = "github.com/lean-delivery/tf-module-static-artifact-storage"
+    
+    region          = "us-west-2"
```
Comment (ToROxI) [inline README.md]: What is the purpose of `region` variable here?

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+  default_tags = {
+    Project     = "${var.project}"
+    Environment = "${var.environment}"
+  }
+
+  acm_certificate_arn = "${var.acm_certificate_arn == "" ? module.aws-cert.acm_certificate_arn : var.acm_certificate_arn}"
+  distribution_label  = "${var.project}-${var.environment}-distribution_label"
+}
+
+module "vpc" {
```
Comment (ToROxI) [inline main.tf]: What the purpose of VPC in this module?

File: README.md (hunk 4 above)
Comment (arman-promo) [reply README.md]: There is no purpose, its mistake, fixed

File: main.tf (hunk 5 above)
Comment (arman-promo) [reply main.tf]: One of requirements is reusing existed security groups, and I have no idea how to link security group with cloudfront and s3

Comment (arman-promo) [reply main.tf]: removed vpc and security groups

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    }
+  }
+
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
+    }
+  }
+
+  viewer_certificate {
```
Comment (ToROxI) [inline main.tf:124]: As far as viewer certificate is mandatory it's better to integrate [SSL certificate creation](https://github.com/lean-delivery/tf-module-aws-acm) inside this module

Comment (arman-promo) [reply main.tf:124]: added [tf-module-aws-acm](https://github.com/lean-delivery/tf-module-aws-acm) module

File: README.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+| whitelist | List of whitelist CIDRs | list | `[]` | no |
+| acm_certificate_arn | ARN of certificate | string | `""` | no |
+| price_class | Default price class | string | `"PriceClass_200"` | no|
+
+## Outputs
+
+| Name | Description |
+|------|-------------|
+| whitelist | List of whitelist CIDRs |
+
```
Comment (ToROxI) [inline README.md:48]: Please add `bucket_name` and `bucket_arn` to output

File: main.tf [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+  }
+
+  default_cache_behavior {
+    allowed_methods  = ["HEAD", "GET", "OPTIONS"]
+    cached_methods   = ["HEAD", "GET", "OPTIONS"]
+    target_origin_id = "${local.distribution_label}"
+
+    min_ttl                = 0
+    default_ttl            = 3600
+    max_ttl                = 86400
```
Comment (ToROxI) [inline main.tf:105]: TTL parameters should be configurable via variables

File: main.tf [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+  metric_name = "WhitelistWafRule"
+
+  predicates {
+    data_id = "${aws_waf_ipset.whitelist_ipset.id}"
+    negated = false
+    type    = "IPMatch"
+  }
+}
+
+resource "aws_waf_web_acl" "whitelist_waf_acl" {
```
Comment (ToROxI) [inline main.tf:159]: WAF should not be mandatory

**Review by ToROxI — CHANGES_REQUESTED** [review]

File: main.tf [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+  tags = "${merge(local.default_tags, var.tags)}"
+}
+
+resource "aws_s3_bucket_policy" "default" {
+  bucket = "${aws_s3_bucket.origin.id}"
+  policy = "${data.template_file.default.rendered}"
+}
+
+module "aws-cert" {
+  source = "github.com/lean-delivery/tf-module-aws-acm"
```
Comment (ToROxI) [inline main.tf:65]: please pin to the specific version

File: variables.tf [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+variable "tags" {
+  type        = "map"
+  default     = {}
+  description = "Additional tags for resources"
+}
+
+variable "root_domain" {
+  default     = "example.com"
+  description = "Name of Route53 zone (if 'create_route53_zone' = True)"
```
Comment (ToROxI) [inline variables.tf:19]: where is `create_route53_zone ` input variable?

**Review by ToROxI — CHANGES_REQUESTED** [review]

**Pull request closed without merging**